### PR TITLE
Annotate histograms with bucket boundaries

### LINKED ISSUE #53359: Add bucket size hints to histograms
Histograms use buckets to efficiently store data. Good data requires good histogram bucket sizes, customized to an instrument. Hacks were required in .NET 8 to be in a good place here.

A feature to [improve this in System.Diaganostics.Metrics](https://github.com/dotnet/runtime/issues/63650) is planned for .NET 9. We need to update our metrics to use the new API where appropriate and remove hacks.

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspnetcore/issues/53359

### CODE REVIEW

**Comment by JamesNK** [comment]
cc @noahfalk @tarekgh

**Review by Kielek — APPROVED** [review]

File: src/Shared/Metrics/MetricsConstants.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http;
+
+internal static class MetricsConstants
+{
+    // Follows boundaries from http.server.request.duration/http.client.request.duration
+    public static readonly IReadOnlyList<double> ShortSecondsBucketBoundaries = [0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10];
+
+    // Not based on a standard. See https://github.com/open-telemetry/semantic-conventions/issues/336
```
Comment (JamesNK) [inline src/Shared/Metrics/MetricsConstants.cs:11]:
```suggestion
    // Not based on a standard. Larger bucket sizes for longer lasting operations, e.g. HTTP connection duration. See https://github.com/open-telemetry/semantic-conventions/issues/336
```

Comment (tarekgh) [reply src/Shared/Metrics/MetricsConstants.cs]: I am seeing the issue still open. Is this means still undecided what are the final boundaries?

Comment (tarekgh) [reply src/Shared/Metrics/MetricsConstants.cs]: I am seeing the issue still open. Is this means still undecided what are the final boundaries?

**Review by tarekgh — APPROVED** [review]
LGTM

File: src/Shared/Metrics/MetricsConstants.cs (hunk 1 above)
Comment (JamesNK) [reply src/Shared/Metrics/MetricsConstants.cs]: Yes

**Review by noahfalk — APPROVED** [review]
LGTM as well (also glad in general to see the advice API fitting smoothly into existing code)

**Comment by noahfalk** [comment]
fyi @samsp-msft

**Comment by chester89** [comment]
Can I ask if this was released? Because [last nuget version for RL package](https://www.nuget.org/packages/Microsoft.AspNetCore.RateLimiting) is 7.0.0-rc...

**Comment by martincostello** [comment]
That package was moved into the ASP.NET Core Shared Framework during the development of .NET 7 and is no longer an independently shipped NuGet package.

**Comment by chester89** [comment]
I see. Thanks

**Pull request merged**